File: baik/Algorithm/lv2/귤 고르기.swift
import Foundation

func solution(_ k:Int, _ tangerine:[Int]) -> Int {
    let max = tangerine.max()!
    var arr : [Int] = Array(repeating:0, count: max)
    for i in tangerine {
        arr[i-1] += 1
    }
    
    
    arr.sort(by: >)
    
    var ans = 0
    var count = 0
    for i in arr {
        if count < k {
            count += i
            ans += 1
        } else {
            break
        }
    }
    return ans
}

File: im/프로그래머스/프로그래머스 Lv2/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {

        List<Integer> arrList = new ArrayList<>();

        Arrays.sort(tangerine);

        int count = 1;
        for(int i = 1; i < tangerine.length; i++){
            if(tangerine[i] == tangerine[i-1])count++;
            else{
                arrList.add(count);
                count = 1;
            }

            if(i == tangerine.length-1) arrList.add(count);
        }


        int[] countArr = new int[arrList.size()];
        for(int i =0; i < countArr.length; i++){
            countArr[i] = arrList.get(i);
        }

        Arrays.sort(countArr);

        int sum = 0;

        for(int i = 0; i<countArr.length; i++){
            sum += countArr[countArr.length-1 - i];
            if(sum >= k) return i+1;
        }

        return tangerine.length;
    }
}